src/Planners/include/BipedalLocomotion/Planners/UnicycleTrajectoryPlanner.h

```diff
@@ -0,0 +1,234 @@
+{
+class UnicycleTrajectoryPlanner;
+struct UnicycleTrajectoryPlannerInput;
+struct UnicycleTrajectoryPlannerOutput;
+struct UnicycleTrajectoryPlannerParameters;
+} // namespace BipedalLocomotion::Planners
+
+struct BipedalLocomotion::Planners::UnicycleTrajectoryPlannerInput
+{
+    /*
+    if UnicycleController::PERSON_FOLLOWING, the plannerInput is a vector of size 2 (i.e., [x, y])
+    if UnicycleController::DIRECT, the plannerInput is a vector of size 3 (i.e., [xdot, ydot, wz])
+    */
+    Eigen::VectorXd plannerInput; // The input to the unicycle planner.
+
+    DCMInitialState dcmInitialState; // The initial state of the DCM trajectory generator.
+
+    struct COMInitialState
+    {
+        Eigen::Vector2d initialPlanarPosition; // The initial planar position of the CoM.
+        Eigen::Vector2d initialPlanarVelocity; // The initial planar velocity of the CoM.
+    };
+
+    COMInitialState comInitialState; // The initial state of the CoM.
+
+    bool isLeftLastSwinging; // True if the last foot that was swinging is the left one. False
+                             // otherwise.
+
+    double initTime; // The initial time of the trajectory.
+
+    manif::SE3d measuredTransform; // The measured transform of the last foot that touched
+                                   // the floor.
+
+    static UnicycleTrajectoryPlannerInput generateDummyUnicycleTrajectoryPlannerInput();
+};
+
+struct BipedalLocomotion::Planners::UnicycleTrajectoryPlannerOutput
+{
+    struct COMTrajectory
+    {
+        std::vector<Eigen::Vector3d> position;
+        std::vector<Eigen::Vector3d> velocity;
+        std::vector<Eigen::Vector3d> acceleration;
+    };
+
+    struct DCMTrajectory
+    {
+        std::vector<Eigen::Vector2d> dcmPosition;
+        std::vector<Eigen::Vector2d> dcmVelocity;
+    };
+
+    struct ContactStatus
+    {
+        std::vector<bool> leftFootInContact; // True if the left foot is in contact. False
+                                             // otherwise.
+        std::vector<bool> rightFootInContact; // True if the right foot is in contact. False
+                                              // otherwise.
+        std::vector<bool> UsedLeftAsFixed; // True if the left foot is the last that got in contact.
+    };
+
+    struct Steps
+    {
+        std::deque<Step> leftSteps, rightSteps;
+    };
+
+    COMTrajectory comTrajectory; // The CoM trajectory;
+
+    DCMTrajectory dcmTrajectory; // The DCM trajectory;
+
+    ContactStatus contactStatus; // The contact status of the feet;
+
+    Contacts::ContactPhaseList ContactPhaseList; // The list of foot contact phases;
+
+    Steps steps; // The list of steps and their phases;
+
+    std::vector<size_t> mergePoints; // Indexes of the merge points of the trajectory;
+};
+
+struct BipedalLocomotion::Planners::UnicycleTrajectoryPlannerParameters
+{
+    double dt; // The sampling time of the planner.
+
+    double plannerHorizon; // The time horizon of the planner.
```

Comment:
In the project starting from https://github.com/ami-iit/bipedal-locomotion-framework/pull/630 and in  #702 we started using   `std::chrono::nanoseconds` for expressing the time
```suggestion
	std::chrono::nanoseconds dt; // The sampling time of the planner.

    std::chrono::nanoseconds plannerHorizon; // The time horizon of the planner.
```

Reply:
makes sense to me! Then, I will probably need to cast it back to double when passing time values to the underlying `unicycle-footsteps-planner`.

Reply:
yes!

Reply:
done with [eb87904](https://github.com/ami-iit/bipedal-locomotion-framework/pull/844/commits/eb87904f17030eaae9d1a10f348e4aceb5236645)